model/modelprocessor/config.go

```diff
@@ -0,0 +1,56 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package modelprocessor
+
+import (
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// ConfigOption allows passing a functional option when creating a new
+// processor
+type ConfigOption func(config) config
+
+// config allow processors to get optional arguments.
+// The config struct is shared amongst every processor using it, but they may
+// not each use every option.
+type config struct {
+	TracerProvider trace.TracerProvider
```

Comment:
I'm happy to make it private (though since the struct is private, it's also private de facto), but I think it should be named `tracerProvider`, not `traceProvider`, to match the interface name.